src/t8_geometry/t8_geometry_implementations/t8_geometry_occ.cxx

```diff
@@ -163,21 +170,538 @@ t8_geometry_occ::t8_geom_evalute_jacobian (t8_cmesh_t cmesh,
-                                                gtreeid);
-  faces = (const int *) t8_cmesh_get_attribute (cmesh, t8_get_package_id (),
-                                                T8_CMESH_OCC_FACE_ATTRIBUTE_KEY,
-                                                gtreeid);
-  T8_ASSERT (edges != NULL);
-  T8_ASSERT (faces != NULL);
-}
-
+inline void
+t8_geometry_occ::t8_geom_load_tree_data (t8_cmesh_t cmesh,
+                                         t8_gloidx_t gtreeid)
+{
+  t8_geometry_w_vertices::t8_geom_load_tree_data (cmesh, gtreeid);
+  edges = (const int *) t8_cmesh_get_attribute (cmesh, t8_get_package_id (),
+                                                T8_CMESH_OCC_EDGE_ATTRIBUTE_KEY,
+                                                gtreeid);
+  faces = (const int *) t8_cmesh_get_attribute (cmesh, t8_get_package_id (),
+                                                T8_CMESH_OCC_FACE_ATTRIBUTE_KEY,
+                                                gtreeid);
+  T8_ASSERT (edges != NULL);
+  T8_ASSERT (faces != NULL);
+}
+
+void
+t8_geometry_occ::t8_geom_evaluate_occ_triangle (t8_cmesh_t cmesh,
+                                                t8_gloidx_t gtreeid,
+                                                const double *ref_coords,
+                                                double out_coords[3]) const
+{
+  T8_ASSERT (active_tree_class == T8_ECLASS_TRIANGLE);
+
+  const int           num_edges = t8_eclass_num_edges[active_tree_class];
+  gp_Pnt              pnt, converted_edge_surface_pnt,
+    interpolated_edge_surface_pnt, interpolated_surface_pnt;
+  Handle_Geom_Curve   curve;
+  Handle_Geom_Surface surface;
+  Standard_Real       first, last;
+  double              interpolated_surface_parameters[2];
+
+  /*
+   * Reference Space    |      Global Space
+   *      [0,1]^2       |         [x,y]^2
+   *                    |
+   *              2     |            2
+   *            / |     |           / \
+   *           /  |     |          /    \
+   *          /   |     |         /       \              o ref_coords in reference space
+   *         /    |     |        /          \            @ glob_ref_coords in global space
+   *        E1     E0   |       E1           E0
+   *       /      |     |      /                \
+   *      /       o     |     /                  @
+   *     /        |     |    /                   |   y
+   *    /         |     |   /                   /    |
+   *   0----E2----1     |  0---------E2--------1     x--x
+   * 
+   */
+
+  /* Linear mapping from ref_coords to out_coords */
+  t8_geom_compute_linear_geometry (active_tree_class, active_tree_vertices,
+                                   ref_coords, out_coords);
+
+  /* Check if face has a linked geometry */
+  if (*faces > 0) {
+    for (int i_edge = 0; i_edge < num_edges; i_edge++) {
+      /* If face carries a surface, edges can't carry surfaces too */
+      T8_ASSERT (edges[i_edge + num_edges] == 0);
+    }
```

Comment:
This should be put in a `#ifdef T8_ENABLE_DEBUG`. This way the loop is not executed in release mode